squad/plugins/linux_log_parser.py

```diff
@@ -1,67 +1,87 @@
-            self.__log__.append(line)
-            if self.max_lines and len(self.__log__) >= self.max_lines:
-                self.done = True
+ONELINERS = [
+    ('check-kernel-oops', r'^[^\n]+Oops(?: -|:).*?$'),
+    ('check-kernel-fault', r'^[^\n]+Unhandled fault.*?$'),
+    ('check-kernel-warning', r'^[^\n]+WARNING:.*?$'),
+]
 
-    @property
-    def log(self):
-        return "\n".join(self.__log__)
-
-
-class Oops(Issue):
-    name = 'oops'
-    max_lines = 50
-
-    def __feed__(self, line):
-        if "------------[ cut here ]------------" in line:
-            self.found = True
-
-
-class KernelPanic(Issue):
-    name = 'kernel-panic'
-
-    def __feed__(self, line):
-        if 'Kernel panic - not syncing:' in line:
-            self.found = True
-        if '---[ end Kernel panic - not syncing' in line:
-            self.done = True
+# Tip: broader regexes should come first
+REGEXES = MULTILINERS + ONELINERS
 
 
 class Plugin(BasePlugin):
+    def __compile_regexes(self, regexes):
+        combined = [r'(%s)' % r[REGEX_BODY] for r in regexes]
+        return re.compile(r'|'.join(combined), re.S | re.M)
+
+    def __kernel_msgs_only(self, log):
+        kernel_msgs = re.findall(r'^\[[ \d]+\.[ \d]+\] .*?$', log, re.S | re.M)
+        return '\n'.join(kernel_msgs)
+
+    def __join_matches(self, matches, regexes):
+        """
+            group regex in python are returned as a list of tuples which each
+            group match in one of the positions in the tuple. Example:
+            regex = r'(a)|(b)|(c)'
+            matches = [
+                ('match a', '', ''),
+                ('', 'match b', ''),
+                ('match a', '', ''),
+                ('', '', 'match c')
+            ]
+        """
+        snippets = {regex_id: [] for regex_id in range(len(regexes))}
+        for match in matches:
+            for regex_id in range(len(regexes)):
+                if len(match[regex_id]) > 0:
+                    snippets[regex_id].append(match[regex_id])
+        return snippets
+
+    def __create_test(self, testrun, suite, test_name, lines):
+        testrun.tests.create(
+            suite=suite,
+            name=test_name,
+            result=(len(lines) == 0),
+            log='\n'.join(lines),
+        )
 
     def postprocess_testrun(self, testrun):
-        if testrun.log_file is not None:
-            project = testrun.build.project
-
-            suite, _ = project.suites.get_or_create(slug='linux-log-parser')
-            for issue in Issue.find(testrun.log_file):
-                testrun.tests.create(
-                    suite=suite,
-                    name='check-' + issue.name,
-                    result=(not issue.found),
-                    log=issue.log
-                )
+        log = self.__kernel_msgs_only(testrun.log_file)
+        suite, _ = testrun.build.project.suites.get_or_create(slug='linux-log-parser')
+        test_name_suffix = ('-%s' % testrun.job_id) if testrun.job_id else ''
```

Comment:
I would use the job name instead. this way if the jobs are named correctly, then you have recurring test names across builds.

Reply:
Makes sense. I remember a while ago @mwasilew saying that this could also be a field in the metadata. At any rate, I've changed `postprocess_testrun` to `postprocess_testjob` so the plugin would have easy access to the testjob.